### PR TITLE
Clean up some unused ember methods from codegen & headers

### PR DESCRIPTION
**Pull request by andy31415**
There are a few methods which seem to be codegened as empty implementations and callbacks, however they never seem used inside our src. These seem ember legacy, so cleaning them up to have simpler codegen and simpler code overall.

### CODE REVIEW

File: src/app/util/af.h [hunk 1]
```diff
@@ -467,125 +439,17 @@ void emberReverseMemCopy(uint8_t * dest, const uint8_t * src, uint16_t length);
...
 
-/**
- * @brief Returns the current network parameters.
- */
-EmberStatus emberAfGetNetworkParameters(EmberNodeType * nodeType, EmberNetworkParameters * parameters);
-
-/**
- * @brief Returns the current node type.
- */
-EmberStatus emberAfGetNodeType(EmberNodeType * nodeType);
```
Comment (bzbarsky-apple) [inline src/app/util/af.h:515]: Is the EmberNodeType type unused after this?  Followup to remove it, please?

File: src/app/util/attribute-storage.cpp [hunk 2]
```diff
@@ -699,25 +699,12 @@ uint8_t emberAfClusterIndex(EndpointId endpoint, ClusterId clusterId, EmberAfClu
...
 // Returns whether the given endpoint has the server of the given cluster on it.
 bool emberAfContainsServer(EndpointId endpoint, ClusterId clusterId)
 {
     return (emberAfFindCluster(endpoint, clusterId, CLUSTER_MASK_SERVER) != nullptr);
 }
 
-// Returns whether the given endpoint has the client of the given cluster on it.
-bool emberAfContainsClient(EndpointId endpoint, ClusterId clusterId)
-{
-    return (emberAfFindCluster(endpoint, clusterId, CLUSTER_MASK_CLIENT) != nullptr);
```
Comment (bzbarsky-apple) [inline src/app/util/attribute-storage.cpp:718]: Followup: I think after this all non-dead call-sites of `emberAfFindCluster` pass `CLUSTER_MASK_SERVER`.  In particular, I think `emberAfClusterDefaultResponseCallback` and `emberAfClusterMessageSentCallback` are dead code and should be removed.

Once that happens, we should just remove that argument and rename the function to `emberAfFindServerCluster`...

**Review by bzbarsky-apple — APPROVED** [review]

File: src/app/util/af.h (hunk 1 above)
Comment (andy31415) [reply src/app/util/af.h:515]: Removed it and EmberChildData from type_stubs.h.

File: src/app/util/attribute-storage.cpp (hunk 2 above)
Comment (andy31415) [reply src/app/util/attribute-storage.cpp:718]: I will do the removal of CLUSTER_MASK_SERVER as a followup. For now just deleting unused code.

Did remove the emberAfClusterDefaultResponseCallback and emberAfClusterMessageSentCallback

**Review by turon — APPROVED** [review]

**Pull request merged**